data/scenarios/Challenges/Ranching/gated-paddock/meandering-sheep.sw

```diff
@@ -0,0 +1,36 @@
+// A "sheep" that wanders around randomly.
+
+let forever : cmd unit -> cmd unit = \c. c ; forever c in
+let repeat : int -> cmd unit -> cmd unit =
+  \n. \c. if (n == 0) {} {c ; repeat (n-1) c} in
+let randdir : cmd dir =
+  d <- random 4;
+  return (
+    if (d == 0) {north}
+    {if (d == 1) {east}
+    {if (d == 2) {south} {west}}})
+  in
+
+forever (
+  n <- random 30;
+  wait (30 + n);
+  d <- randdir;
+  turn d;
+  dist <- random 3;
+  repeat dist (
+    b <- blocked;
+    if b {} {
+      move;
+
+      // Sheep can drown.
+      hasWater <- ishere "water";
+      if hasWater {
+        say "whoops!";
```

Comment:
Oh, so that is how drowning sheep sound. 😆  🌊 🐑 🌊 

As an alternative (e.g. if sheep were normal robots and just drowned) a referee robot could check if the sheep still exist and say something if they were not found.